platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/PbjStreamHasher.java

```diff
@@ -64,32 +66,43 @@ public PlatformEvent hashEvent(@NonNull final PlatformEvent event) {
      * @param event the event to hash
      */
     public void hashUnsignedEvent(@NonNull final UnsignedEvent event) {
-        final Hash hash = hashEvent(event.getEventCore(), event.getTransactions());
+        final Hash hash = hashEvent(event.getEventCore(), event.getTransactions(), true);
         event.setHash(hash);
     }
 
     /**
      * Hashes the given event and returns the hash.
      *
-     * @param eventCore the event to hash
-     * @param transactions the transactions to hash
-     *
+     * @param eventCore         the event to hash
+     * @param transactions      the transactions to hash
      * @return the hash of the event
      */
     @NonNull
-    private Hash hashEvent(@NonNull final EventCore eventCore, @NonNull final List<TransactionWrapper> transactions) {
+    private Hash hashEvent(
+            @NonNull final EventCore eventCore,
+            @NonNull final List<TransactionWrapper> transactions,
+            final boolean isNewFormat) {
         try {
             EventCore.PROTOBUF.write(eventCore, eventStream);
             for (final TransactionWrapper transaction : transactions) {
-                EventTransaction.PROTOBUF.write(transaction.getTransaction(), transactionStream);
-                byte[] hash = transactionDigest.digest();
-                transaction.setHash(Bytes.wrap(hash));
-                eventStream.writeBytes(hash);
+                if (isNewFormat) {
+                    transactionStream.writeBytes(Objects.requireNonNull(transaction.getApplicationTransaction()));
+                    processTransactionHash(transaction);
+                } else {
+                    EventTransaction.PROTOBUF.write(transaction.getTransaction(), transactionStream);
+                    processTransactionHash(transaction);
```

Comment:
this method is repeated also in line 90. It can be extracted outside of the if/else block